lib/pure/typetraits.nim

```diff
@@ -15,6 +15,24 @@
 import std/private/since
 export system.`$` # for backward compatibility
 
+type SomeSparseEnum* = (not Ordinal) and enum ## sparse enum's, a.k.a enum with holes
```

Comment:
This should be `HoleyEnum` or `EnumWithHoles`; you wouldn't call an enum with only one hole sparse.

Reply:
Holey enum seems to be a very non standard terminology that almost only nim uses though. On the other hand, I've seen sparse enum used (with the same meaning) in a number of places, and sparse vs dense (eg indexes etc) is also commonly used for the same concept. Whether there's just 1 hole or many holes shouldn't matter; you'd have to treat it differently from regular enum's.

Reply:
`AdjoinedEnum` or  `ContiguousEnum` have my vote, since we're bikeshedding names.

Reply:
Hmm, yeah, sparse enum is more common outside of nim. I think we should do a vote to get and idea which terminology nimions would prefer :)